pkg/registry/registrytest/etcd.go

```diff
@@ -208,82 +189,33 @@ func (t *Tester) setObject(ctx api.Context, obj runtime.Object) error {
 	if err != nil {
 		return err
 	}
-	key = etcdtest.AddPrefix(key)
-
-	codec, err := getCodec(obj)
-	if err != nil {
-		return err
-	}
-	_, err = t.fakeClient.Set(key, runtime.EncodeOrDie(codec, obj), 0)
-	return err
+	return t.storage.Storage.Set(ctx, key, obj, nil, 0)
 }
 
 func (t *Tester) setObjectsForList(objects []runtime.Object) []runtime.Object {
-	result := make([]runtime.Object, len(objects))
-	key := etcdtest.AddPrefix(t.storage.KeyRootFunc(t.tester.TestContext()))
-
-	if len(objects) > 0 {
-		nodes := make([]*etcd.Node, len(objects))
-		for i, obj := range objects {
-			codec, _ := getCodec(obj)
-			encoded := runtime.EncodeOrDie(codec, obj)
-			decoded, _ := codec.Decode([]byte(encoded))
-			nodes[i] = &etcd.Node{Value: encoded}
-			result[i] = decoded
-		}
-		t.fakeClient.Data[key] = tools.EtcdResponseWithError{
-			R: &etcd.Response{
-				Node: &etcd.Node{
-					Nodes: nodes,
-				},
-			},
-			E: nil,
-		}
-	} else {
-		t.fakeClient.Data[key] = tools.EtcdResponseWithError{
-			R: &etcd.Response{},
-			E: t.fakeClient.NewError(tools.EtcdErrorCodeNotFound),
-		}
+	key := t.storage.KeyRootFunc(t.tester.TestContext())
+	if err := storagetesting.CreateObjList(key, t.storage.Storage, objects); err != nil {
+		return nil
 	}
-	return result
-}
-
-func (t *Tester) setResourceVersion(resourceVersion uint64) {
-	t.fakeClient.ChangeIndex = resourceVersion
-}
-
-func (t *Tester) injectWatchError(err error) {
-	t.fakeClient.WatchInjectError <- err
+	return objects
 }
 
 func (t *Tester) emitObject(obj runtime.Object, action string) error {
-	codec, err := getCodec(obj)
-	if err != nil {
-		return err
-	}
-	encoded, err := codec.Encode(obj)
-	if err != nil {
-		return err
-	}
-	node := &etcd.Node{
-		Value: string(encoded),
-	}
-	var prevNode *etcd.Node = nil
-	if action == etcdstorage.EtcdDelete {
-		prevNode = node
-	}
-	t.fakeClient.WatchResponse <- &etcd.Response{
-		Action:   action,
-		Node:     node,
-		PrevNode: prevNode,
+	ctx := t.tester.TestContext()
+	var err error
+
+	switch action {
+	case etcdstorage.EtcdCreate:
```

Comment:
So before there was a CAS, and there exists other Update tests, as you can see.  The watch emitter imho is kind of a farce to begin with, b/c it's not really testing what it may imply here.